arangosh/Restore/RestoreFeature.cpp

```diff
@@ -1784,4 +1824,68 @@ Result RestoreFeature::getFirstError() const {
   return {TRI_ERROR_NO_ERROR};
 }
 
+void RestoreFeature::ProgressTracker::updateStauts(const std::string& filename,
+                                                   const RestoreFeature::CollectionStatus& status) {
+  std::unique_lock guard(_collectionStatesMutex);
+  _collectionStates[filename] = status;
+
+  // TODO currently every change is immediately written to disk
+  // maybe we can collect multiple changes and write only once.
+  VPackBuilder builder;
+  {
+    VPackObjectBuilder object(&builder);
+
+    for (auto const& [filename, state] : _collectionStates) {
+      builder.add(VPackValue(filename));
+      state.toVelocyPack(builder);
+    }
+  }
+  LOG_DEVEL << "Continuation file = " << builder.toJson();
+
+  auto file = directory.writableFile("continue.json", true, 0, false);
+  file->spit(builder.toJson());
+}
+
+RestoreFeature::CollectionStatus RestoreFeature::ProgressTracker::getStatus(const std::string& filename) {
+  std::shared_lock guard(_collectionStatesMutex);
+  return _collectionStates[filename];
```

Comment:
when fetching the status for a collection which is not in `_collectionStates`, this will magically work, as it will return a default-constructed `CollectionStatus` object with state `UNKNOWN`.
So all good, but there should be a comment here indicating that it works also for unknown collections.

Reply:
Okay, this was absolutely intended. But I will add a comment, yes.